bundles/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java

```diff
@@ -715,6 +739,23 @@ private JSONObject getJSONHeaders(HttpFields httpFields) {
         return headersJSON;
     }
 
+    private JSONArray createActionArray(@Nullable String... actionStrings) {
+        JSONArray actionArray = new JSONArray();
+        for (String actionString : actionStrings) {
+            if (actionString == null) {
+                continue;
+            }
+            String[] parts = actionString.split("=", 2);
```

Comment:
The action part is guaranteed/required to not contain any `=`? (I am thinking among the lines of `parts` potentially having more than 2 elements). It may be safer to use `indexOf('=')` and `substring()`.

Reply:
hmm , maybe I'm not following,  but  passing a limit of 2 ensures it only splits into 2 parts, matching on the first occurrence of '=',  so it would ignore anything in the action part.... or am i misunderstanding your feedback ?

Reply:
> passing a limit of 2 ensures it only splits into 2 parts, matching on the first occurrence of '='

I haven't noticed the two, then I'am fine with it :+1:

Reply:
I hadn't noticed the 2 either, should have paid more attention.